spec/support/shared_examples_for_network_errors_during_send.rb

```diff
@@ -0,0 +1,37 @@
+# -*- encoding : utf-8 -*-
+require 'spec_helper'
+
+shared_examples_for 'NetworkSendErrors' do
+
+  describe 'handles a network error during message sending' do
+
+    before do
+      allow_any_instance_of(ActionMailer::MessageDelivery).
+        to receive(:deliver_now).
+          and_raise(Errno::ETIMEDOUT)
+
+      send_request
+    end
+
+    it 'does not send the email' do
+      deliveries = ActionMailer::Base.deliveries
+      expect(deliveries.size).to eq(0)
+    end
+
+    it 'sets the described_state of the request to "error_message"' do
+      expect(request.reload.described_state).to eq('error_message')
+    end
+
+    it 'logs a "send_error" event' do
+      event = request.reload.info_request_events.last
+      expect(event.event_type).to eq 'send_error'
+    end
+
+    it 'stores the reason for the failure' do
+      event = request.reload.info_request_events.last
+      expect(event.params[:reason]).to eq 'Connection timed out'
+    end
+
```

Comment:
It also forces the outgoing message to get saved, so should test that. 

```
it 'ensures the outgoing message is persisted' do
  # …
  expect(outgoing_message).to be_persisted
end
```

https://github.com/mysociety/alaveteli/pull/5152/commits/1130fe8a559afa88680c31cb9e5630796ddfab4d#diff-266c99b1323db6c6172f9d5751a56bd8R163

Reply:
Ah, the above is just for followups; in request controller we force the `InfoRequest` to get saved:

https://github.com/mysociety/alaveteli/pull/5152/commits/1130fe8a559afa88680c31cb9e5630796ddfab4d#diff-1fa1a855bf4fb5aaec94a56340bcc062R428

Could put these in the `'a network error occurs while sending a followup'` context of the controller specs?

Reply:
oh, actually 2 different things here:
- both should ensure that our new `outgoing_message` gets persisted so that should go into the shared spec file
- request_controller needs an additional check that the parent `info_request` was updated (there's an earlier `save!` so `persisted?` is the wrong thing here) with the new `outgoing_message` data

Reply:
ugh, that ensure save isn't really needed as (I think) the `info_request` (and relations) get saved during the log event; updating the test to check `is_persisted` and "does the related outgoing message have the expected status?"